app/views/requests/show.html.erb

```diff
@@ -67,31 +67,29 @@
               <tr>
                 <th>Item</th>
                 <th>Quantity</th>
-                <% custom_units = Flipper.enabled?(:enable_packs) &&
-                                  @request.item_requests.any? { |item| item.request_unit } %>
-                <% if custom_units %>
+                <% if @custom_units %>
                   <th>Units (if applicable)</th>
                 <% end %>
-                <% default_storage_location = @request.partner.default_storage_location_id ||
-                                              @request.organization.default_storage_location %>
-                <% if default_storage_location %>
+                <% if @default_storage_location %>
                   <th>Default storage location inventory</th>
                 <% end %>
                 <th>Total Inventory</th>
               </tr>
               </thead>
               <tbody>
-              <% @request_items.each do |item| %>
+              <% @item_requests.each do |item_request| %>
                 <tr>
-                  <td><%= item.name %></td>
-                  <td><%= item.quantity %></td>
-                  <% if custom_units %>
-                    <td><%= item.unit&.pluralize(item.quantity) %></td>
+                  <td><%= item_request.item&.name || item_request.name %></td>
```

Comment:
Tries to look up the "current" item name, and then falls back to the cached item_request.name. We do this elsewhere also.